File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset = set(nums)
        length = 0
        for num in numset:
            if num - 1 not in numset:
                curr_length = 0
                while num in numset:
                    curr_length += 1
                    num = num + 1
                length = max(curr_length, length)
        return length

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq_count;
        int max_freq = 0;
        for (int num : nums) {
            freq_count[num]++;
            max_freq = max(max_freq, freq_count[num]);
        }

        vector<vector<int>> freq(max_freq, vector<int>());
        for (const auto& pp : freq_count) {
            int n = pp.first, fr = pp.second;
            freq[fr - 1].push_back(n);
        }

        int pointer = max_freq - 1;
        int curr = 0;
        vector<int> res(k);
        while (curr < k) {
            for (int num : freq[pointer]) {
                res[curr] = num;
                curr++;
            }
            pointer--;
        }
        return res;
    }
};
